src/components/TeacherViews/LessonMedia/LessonMedia.view.tsx

```diff
@@ -38,15 +58,150 @@ type LessonMediaProps =
       lesson: NonCanonicalLesson;
     };
 
-export function LessonMedia(props: LessonMediaProps) {
+export const LessonMedia = (props: LessonMediaProps) => {
   const { isCanonical, lesson } = props;
-  const { lessonTitle, lessonSlug, keyStageTitle } = lesson;
+  const { lessonTitle, lessonSlug, keyStageTitle, mediaClips } = lesson;
 
   const commonPathway = getCommonPathway(
     props.isCanonical ? props.lesson.pathways : [props.lesson],
   );
 
-  const { programmeSlug, unitSlug, subjectTitle } = commonPathway;
+  const { programmeSlug, unitSlug, subjectTitle, yearTitle } = commonPathway;
+
+  const router = useRouter();
+  const { query } = router;
+  const { video } = query;
+
+  // construct list of all clips in one array
+  const listOfAllClips = Object.keys(mediaClips)
+    .map((learningCycle) =>
+      mediaClips[learningCycle as LearningCycle].map(
+        (mediaClip: MediaClip) => mediaClip,
+      ),
+    )
+    .flat();
+
+  const [currentClip, setCurrentClip] = useState(
+    getInitialCurrentClip(listOfAllClips, video),
+  );
+  const [currentIndex, setCurrentIndex] = useState(
+    currentClip ? listOfAllClips.indexOf(currentClip) : 0,
+  );
+
+  const playedVideosList: string[] =
+    sessionStorage.getItem("playedVideosList")?.split(",") || [];
+
+  // action performed on media clip item click
+  const onMediaClipClick = (clipSlug: string) => {
+    const clickedMediaClip = listOfAllClips.find(
+      (clip) => clip.slug === clipSlug,
+    );
+    setCurrentClip(clickedMediaClip);
+    setCurrentIndex(
+      clickedMediaClip ? listOfAllClips.indexOf(clickedMediaClip) : 0,
+    );
+
+    // add video parameter to the url
+    query.video = clipSlug;
+    router.replace(router);
+  };
+
+  const handleVideoPlayed = (event: VideoEventCallbackArgs) => {
+    if (event.event === "play") {
+      if (currentClip && !playedVideosList.includes(currentClip?.slug)) {
+        // add played video to session storage
+        const updatedPlayedVideosList = [
+          ...playedVideosList,
+          currentClip.slug,
+        ].toString();
+        sessionStorage.setItem("playedVideosList", updatedPlayedVideosList);
+      }
+    }
+  };
+
+  const videoPlayer = currentClip && (
+    <VideoPlayer
+      playbackId={getPlaybackId(currentClip)}
+      playbackPolicy={"signed"}
+      title={currentClip.mediaClipTitle}
+      location={"lesson"}
+      isLegacy={false}
+      userEventCallback={handleVideoPlayed}
+    />
+  );
+
+  // media clip list component
+  const mediaClipList = (
+    <OakMediaClipList
+      lessonTitle={lessonTitle}
+      currentClipCounter={currentIndex + 1}
+      totalClipCounter={listOfAllClips.length}
+    >
+      {listOfAllClips.map((mediaClip: MediaClip, index: number) => {
+        const {
+          videoObject,
+          mediaObject,
+          mediaClipTitle,
+          learningCycleTitle,
+          slug,
+          mediaType,
+        } = mediaClip;
+
```

Comment:
Not something to change now, but hopefully when we have the data we can refactor the video and media clips into one

Reply:
agreed!